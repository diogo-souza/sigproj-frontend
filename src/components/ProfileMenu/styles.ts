import styled, { css } from 'styled-components';
import { NavLink as NavLinkRRD } from 'react-router-dom';

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    border-bottom: 0.1rem solid ${theme.colors.lightGray};

    @media (min-width: 768px) {
      flex-direction: column;
      border: 0;
      a:not(:last-child) {
        border-bottom: 0.1rem solid ${theme.colors.lightGray};
      }
    }
  `}
`;

export const NavLink = styled(NavLinkRRD)`
  ${({ theme, activeClassName }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.titleBlack};
    text-decoration: none;
    display: flex;
    align-items: center;
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    transition: background, color, ${theme.transition.default};
    &:hover {
      background: ${theme.colors.primary};
      color: ${theme.colors.titleWhite};
    }
    > span {
      margin-left: ${theme.spacings.xsmall};
    }
    @media (max-width: 768px) {
      width: min(24px, 100%);
      justify-content: center;
      flex: 1;
      > span {
        display: none;
      }
    }
    &.${activeClassName} {
      background: ${theme.colors.primary};
      color: ${theme.colors.titleWhite};
    }
  `}
`;
