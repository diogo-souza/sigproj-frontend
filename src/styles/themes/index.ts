export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem',
  },
  border: {
    radius: '0.4rem',
  },
  font: {
    family:
      "Open-Sans, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    light: 300,
    regular: 400,
    semiBold: 500,
    bold: 600,
    extraBold: 800,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem',
    },
  },
  colors: {
    primary: '#B51222',
    secondary: '#159CAE',
    tertiary: '#157BAE',

    success: '#48BB78',
    danger: '#dc3545',
    dangerShadow: 'rgba(220, 53, 69, 0.4)',
    warning: '#ffc107',
    focus: '#0d6efd',
    focusShadow: 'rgba(75, 174, 227, 0.4)',

    mainBg: '#F4F5F7',
    minorBg: '#242020',

    black: '#171923',
    darkGray: '#212227',
    gray: '#939393',
    lightGray: '#E1E1E1',
    white: '#FFFFFF',

    titleBlack: '#171923',
    titleWhite: '#FFFFFF',
    bodyBlack: '#939393',
    bodyWhite: '#FFFFFF',
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
  },
  transition: {
    default: '0.3s ease-in-out',
    fast: '0.15s ease-in-out',
  },
} as const;
