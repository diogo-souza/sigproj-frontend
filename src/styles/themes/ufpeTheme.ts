export default {
  border: {
    radius: '0.4rem',
  },
  font: {
    family:
      "Open-Sans, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    light: 300,
    regular: 400,
    semiBold: 500,
    bold: 600,
    extraBold: 800,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem',
    },
  },
  colors: {
    primary: '#B51222',
    secondary: '#159CAE',
    tertiary: '#157BAE',
    mainBg: '#FAFAFA',
    minorBg: '#242020',
    white: '#FFFFFF',
    title: '#2b2b2b',
    text: '#595959',
    highlight: '#007bff',
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
  },
};
