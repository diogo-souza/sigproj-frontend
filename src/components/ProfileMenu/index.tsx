import React from 'react';
import {
  FaUserCircle as UserIcon,
  FaLock as LockIcon,
  FaMapMarkerAlt as AddressIcon,
  FaGraduationCap as GraduationIcon,
} from 'react-icons/fa';
import * as S from './styles';

export type ProfileMenuProps = {
  activeLink?:
    | '/perfil/eu'
    | '/perfil/senha'
    | '/perfil/endereco'
    | '/perfil/ensino';
};

const ProfileMenu: React.FC<ProfileMenuProps> = ({
  activeLink,
}: ProfileMenuProps) => (
  <S.Nav>
    <S.Link
      isActive={activeLink === '/perfil/eu'}
      to="/perfil/eu"
      title="Meu perfil"
    >
      <UserIcon size={24} />
      <span>Meu perfil</span>
    </S.Link>
    <S.Link
      isActive={activeLink === '/perfil/senha'}
      to="/perfil/senha"
      title="Minha senha"
    >
      <LockIcon size={24} />
      <span>Minha senha</span>
    </S.Link>
    <S.Link
      isActive={activeLink === '/perfil/endereco'}
      to="/perfil/endereco"
      title="Meu endereço"
    >
      <AddressIcon size={24} />
      <span>Meu endereço</span>
    </S.Link>
    <S.Link
      isActive={activeLink === '/perfil/ensino'}
      to="/perfil/escolaridade"
      title="Minha escolaridade"
    >
      <GraduationIcon size={24} />
      <span>Minha escolaridade</span>
    </S.Link>
  </S.Nav>
);

export default ProfileMenu;
