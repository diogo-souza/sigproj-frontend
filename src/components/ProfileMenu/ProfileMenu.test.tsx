import { fireEvent, screen } from '@testing-library/react';
import themes from 'styles/themes';
import { renderWithTheme } from 'utils/tests/helpers';

import ProfileMenu from '.';

describe('<ProfileMenu />', () => {
  it('should render the menu', () => {
    const { container } = renderWithTheme(<ProfileMenu />);
    expect(
      screen.getByRole('link', { name: /meu perfil/i }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /minha senha/i }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /Meu endereÃ§o/i }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /Minha escolaridade/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it(`should render the menu with an active link defined`, () => {
    renderWithTheme(<ProfileMenu />);
    screen.getByRole('navigation');
    fireEvent.click(screen.getByRole('link', { name: /meu perfil/i }));
    expect(screen.getByRole('link', { name: /meu perfil/i })).toHaveStyle({
      background: themes.colors.primary,
      color: themes.colors.titleWhite,
    });
    expect(screen.getByRole('link', { name: /minha senha/i })).toHaveStyle({
      background: themes.colors.white,
      color: themes.colors.titleBlack,
    });
  });
});
