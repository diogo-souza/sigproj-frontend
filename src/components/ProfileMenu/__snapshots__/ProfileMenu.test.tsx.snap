// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileMenu /> should render the menu 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 0.1rem solid #E1E1E1;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.6rem 2.4rem;
  -webkit-transition: background,color,0.3s ease-in-out;
  transition: background,color,0.3s ease-in-out;
  background: #FFFFFF;
  color: #171923;
}

.c1:hover {
  background: #B51222;
  color: #FFFFFF;
}

.c1 > span {
  margin-left: 1.6rem;
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    border: 0;
  }

  .c0 a:not(:last-child) {
    border-bottom: 0.1rem solid #E1E1E1;
  }
}

@media (max-width:768px) {
  .c1 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }

  .c1 > span {
    display: none;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="/perfil/eu"
    title="Meu perfil"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 496 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
      />
    </svg>
    <span>
      Meu perfil
    </span>
  </a>
  <a
    class="c1"
    href="/perfil/senha"
    title="Minha senha"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      />
    </svg>
    <span>
      Minha senha
    </span>
  </a>
  <a
    class="c1"
    href="/perfil/endereco"
    title="Meu endereço"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 384 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
      />
    </svg>
    <span>
      Meu endereço
    </span>
  </a>
  <a
    class="c1"
    href="/perfil/escolaridade"
    title="Minha escolaridade"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 640 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M622.34 153.2L343.4 67.5c-15.2-4.67-31.6-4.67-46.79 0L17.66 153.2c-23.54 7.23-23.54 38.36 0 45.59l48.63 14.94c-10.67 13.19-17.23 29.28-17.88 46.9C38.78 266.15 32 276.11 32 288c0 10.78 5.68 19.85 13.86 25.65L20.33 428.53C18.11 438.52 25.71 448 35.94 448h56.11c10.24 0 17.84-9.48 15.62-19.47L82.14 313.65C90.32 307.85 96 298.78 96 288c0-11.57-6.47-21.25-15.66-26.87.76-15.02 8.44-28.3 20.69-36.72L296.6 284.5c9.06 2.78 26.44 6.25 46.79 0l278.95-85.7c23.55-7.24 23.55-38.36 0-45.6zM352.79 315.09c-28.53 8.76-52.84 3.92-65.59 0l-145.02-44.55L128 384c0 35.35 85.96 64 192 64s192-28.65 192-64l-14.18-113.47-145.03 44.56z"
      />
    </svg>
    <span>
      Minha escolaridade
    </span>
  </a>
</nav>
`;
